util/chunk/column.go

```diff
@@ -490,6 +490,44 @@ func (c *Column) GetDecimal(rowID int) *types.MyDecimal {
 	return (*types.MyDecimal)(unsafe.Pointer(&c.data[rowID*types.MyDecimalStructSize]))
 }
 
+// EncodeTo appends the column data slice to the buf
+func (c *Column) EncodeTo(buf [][]byte, eType types.EvalType) ([][]byte, error) {
+	n := c.length
+
+	var fixedTypeSize int
+	switch eType {
+	case types.ETInt, types.ETReal:
+		fixedTypeSize = 8
+	case types.ETDecimal:
+		fixedTypeSize = sizeMyDecimal
+	case types.ETDatetime:
+		fixedTypeSize = sizeTime
+	case types.ETDuration:
+		fixedTypeSize = sizeGoDuration
+	default:
+		fixedTypeSize = 0
+	}
+	var NilFlag byte = 0
+	if fixedTypeSize != 0 {
+		for i := 0; i < n; i++ {
+			if c.IsNull(i) {
+				buf[i] = append(buf[i], NilFlag)
+			} else {
+				buf[i] = append(buf[i], c.data[i*fixedTypeSize:(i+1)*fixedTypeSize]...)
+			}
+		}
+	} else {
+		for i := 0; i < n; i++ {
+			if c.IsNull(i) {
+				buf[i] = append(buf[i], NilFlag)
+			} else {
+				buf[i] = append(buf[i], c.data[c.offsets[i]:c.offsets[i+1]]...)
```

Comment:
`group by a, b`

Will the result of  ['xx', 'y'] == ['x', 'xy']?

Reply:
It seems we need to insert delimiters between elements when encoding.

Reply:
We insert a type flag between elements when encoding. Besides, for variable-length types, we extra insert its length between elements.